### PR TITLE
Implement the new navigation concept for Newshub

### PR DESCRIPTION
**Pull request by dzonidoo**
NHUB-485

## Checklist

- [x] This pull request is not adding new forms that use redux
- [x] This pull request is adding missing TypeScript types to modified code segments where it's easy to do so with confidence
- [x] This pull request is replacing `lodash.get` with optional chaining for modified code segments

### CODE REVIEW

File: newsroom/templates/base_layout.html [hunk 1]
```diff
@@ -172,4 +178,18 @@ <h3 class="a11y-only">{{ gettext('Side Navigation') }}</h3>
 {% endblock %}
 
 </body>
+<script>
```
Comment (petrjasek) [inline newsroom/templates/base_layout.html:181]: this code should go probably rather to assets/index.ts and it should use `gettext` for those titles

File: newsroom/templates/base_layout.html [hunk 2]
```diff
@@ -68,10 +68,16 @@ <h3 class="a11y-only">{{ gettext('Main Navigation Bar') }}</h3>
 
     <div class="contentWrap {% block custom_content_style %}{% endblock %}">
         {% if session.get('name') %}
-        <nav class="{`nav-block ${isNavExpanded ? 'nav-block--pinned' : ''}`}">
+        <nav id="nav" class="nav-block">
             <h3 class="a11y-only">{{ gettext('Side Navigation') }}</h3>
             <ul class="sidenav">
 
+                <li className='d-phone-none'>
+                    <button id="pin-btn" class='nav-block__pin-button' title="Expand" onClick={handleNavExpanded()}>
```
Comment (petrjasek) [inline newsroom/templates/base_layout.html:76]: there should be `gettext` used for title

File: assets/index.ts [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+
+if (pinButton != null) {
+    pinButton.onclick = handleNavExpanded;
+}
+
+if (sessionStorage.getItem('navigation-pinned')) {
+    handleNavExpanded();
+}
+
+function handleNavExpanded() {
```
[the remark this answers is not in the document]
Comment (dzonidoo) [reply assets/index.ts:19]: but this function is not used only once...

[the remark this answers is not in the document]
Comment (tomaskikutis) [reply assets/index.ts:19]: you're right - I missed the second usage. Then only improve the name.

**Review by tomaskikutis — APPROVED** [review]

**Review by petrjasek — APPROVED** [review]

**Pull request merged**